built issue

Run npx --no-install next build
⚠ No build cache found. Please configure build caching for faster rebuilds. Read more: https://nextjs.org/docs/messages/no-cache
Attention: Next.js now collects completely anonymous telemetry regarding usage.
This information is used to shape Next.js' roadmap and prioritize features.
You can learn more, including how to opt-out if you'd not like to participate in this anonymous program, by visiting the following URL:
https://nextjs.org/telemetry

  ▲ Next.js 14.2.13

   Creating an optimized production build ...
Generated 12 documents in .contentlayer
<w> [webpack.cache.PackFileCacheStrategy/webpack.FileSystemInfo] Parsing of /home/runner/work/data-blog/data-blog/embuild-analyses/node_modules/@contentlayer/core/dist/generation/generate-dotpkg.js for build dependencies failed at 'import(filePathJoin(generatedPkgPath, 'generated', 'index.mjs'))'.
<w> Build dependencies behind this expression are ignored and might cause incorrect cache invalidation.
<w> [webpack.cache.PackFileCacheStrategy/webpack.FileSystemInfo] Parsing of /home/runner/work/data-blog/data-blog/embuild-analyses/node_modules/@contentlayer/core/dist/generation/generate-dotpkg.js for build dependencies failed at 'import(filePathJoin(generatedPkgPath, 'generated', 'index.mjs'))'.
<w> Build dependencies behind this expression are ignored and might cause incorrect cache invalidation.
<w> [webpack.cache.PackFileCacheStrategy/webpack.FileSystemInfo] Parsing of /home/runner/work/data-blog/data-blog/embuild-analyses/node_modules/@contentlayer/core/dist/generation/generate-dotpkg.js for build dependencies failed at 'import(filePathJoin(generatedPkgPath, 'generated', 'index.mjs'))'.
<w> Build dependencies behind this expression are ignored and might cause incorrect cache invalidation.
 ✓ Compiled successfully
   Linting and checking validity of types ...
Failed to compile.

./src/components/analyses/bouwondernemers/BouwondernemersDashboard.tsx:340:11
Type error: Type 'TableRow[]' is not assignable to type 'ExportData[]'.
  Type 'TableRow' is missing the following properties from type 'ExportData': label, value

  338 |         <h2 className="text-2xl font-bold">Per subsector</h2>
  339 |         <ExportButtons
> 340 |           data={tableData}
      |           ^
  341 |           title="Bouwondernemers per subsector"
  342 |           slug="bouwondernemers"
  343 |           sectionId="by-sector"
Error: Process completed with exit code 1.

--- a/embuild-analyses/src/components/analyses/bouwondernemers/BouwondernemersDashboard.tsx
+++ b/embuild-analyses/src/components/analyses/bouwondernemers/BouwondernemersDashboard.tsx
@@ -332,12 +332,23 @@ function BySectorSection() {
     return { chartData, tableData }
   }, [selectedRegion])
 
+  const exportData = React.useMemo(() => {
+    return tableData.map((row) => ({
+      label: String(row.periodCells[0] || row.sortValue),
+      value: 0, // Not used for multi-column export
+      periodCells: row.periodCells,
+      ...Object.fromEntries(
+        Object.entries(row).filter(([key]) => key.startsWith('y'))
+      ),
+    }))
+  }, [tableData])
+
   return (
     <div className="space-y-4">
       <div className="flex items-center justify-between">
         <h2 className="text-2xl font-bold">Per subsector</h2>
         <ExportButtons
-          data={tableData}
+          data={exportData}
           title="Bouwondernemers per subsector"
           slug="bouwondernemers"
           sectionId="by-sector"
@@ -458,12 +469,23 @@ function ByGenderSection() {
     return { chartData, tableData }
   }, [selectedRegion])
 
+  const exportData = React.useMemo(() => {
+    return tableData.map((row) => ({
+      label: String(row.periodCells[0] || row.sortValue),
+      value: 0, // Not used for multi-column export
+      periodCells: row.periodCells,
+      ...Object.fromEntries(
+        Object.entries(row).filter(([key]) => key.startsWith('y'))
+      ),
+    }))
+  }, [tableData])
+
   return (
     <div className="space-y-4">
       <div className="flex items-center justify-between">
         <h2 className="text-2xl font-bold">Per geslacht</h2>
         <ExportButtons
-          data={tableData}
+          data={exportData}
           title="Bouwondernemers per geslacht"
           slug="bouwondernemers"
           sectionId="by-gender"
@@ -581,12 +603,23 @@ function ByRegionSection() {
     return { chartData, tableData }
   }, [])
 
+  const exportData = React.useMemo(() => {
+    return tableData.map((row) => ({
+      label: String(row.periodCells[0] || row.sortValue),
+      value: 0, // Not used for multi-column export
+      periodCells: row.periodCells,
+      ...Object.fromEntries(
+        Object.entries(row).filter(([key]) => key.startsWith('y'))
+      ),
+    }))
+  }, [tableData])
+
   return (
     <div className="space-y-4">
       <div className="flex items-center justify-between">
         <h2 className="text-2xl font-bold">Per regio</h2>
         <ExportButtons
-          data={tableData}
+          data={exportData}
           title="Bouwondernemers per regio"
           slug="bouwondernemers"
           sectionId="by-region"
@@ -699,12 +732,23 @@ function ByAgeSection() {
     return { chartData, tableData }
   }, [selectedRegion])
 
+  const exportData = React.useMemo(() => {
+    return tableData.map((row) => ({
+      label: String(row.periodCells[0] || row.sortValue),
+      value: 0, // Not used for multi-column export
+      periodCells: row.periodCells,
+      ...Object.fromEntries(
+        Object.entries(row).filter(([key]) => key.startsWith('y'))
+      ),
+    }))
+  }, [tableData])
+
   return (
     <div className="space-y-4">
       <div className="flex items-center justify-between">
         <h2 className="text-2xl font-bold">Per leeftijd</h2>
         <ExportButtons
-          data={tableData}
+          data={exportData}
           title="Bouwondernemers per leeftijd"
           slug="bouwondernemers"
           sectionId="by-age"

